src/views/UsersList.vue

```diff
@@ -0,0 +1,114 @@
+            <div v-if="user.status == 'FRIENDS'">
+              <svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6 stroke-slate-400" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
+                <path stroke-linecap="round" stroke-linejoin="round" d="M17 20h5v-2a3 3 0 00-5.356-1.857M17 20H7m10 0v-2c0-.656-.126-1.283-.356-1.857M7 20H2v-2a3 3 0 015.356-1.857M7 20v-2c0-.656.126-1.283.356-1.857m0 0a5.002 5.002 0 019.288 0M15 7a3 3 0 11-6 0 3 3 0 016 0zm6 3a2 2 0 11-4 0 2 2 0 014 0zM7 10a2 2 0 11-4 0 2 2 0 014 0z" />
+              </svg>
+            </div>
+            <div v-else-if="user.status == 'NONE'">
+              <svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6 stroke-slate-400" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
+                <path stroke-linecap="round" stroke-linejoin="round" d="M18 9v3m0 0v3m0-3h3m-3 0h-3m-2-5a4 4 0 11-8 0 4 4 0 018 0zM3 20a6 6 0 0112 0v1H3v-1z" />
+              </svg>
+            </div>
+            <div v-else-if="user.status == 'INVITATIONRECEIVED'" class="flex flex-row">
+              <svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6 stroke-slate-400" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
+                <path stroke-linecap="round" stroke-linejoin="round" d="M5 13l4 4L19 7" />
+              </svg>
+              <svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6 stroke-slate-400" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
+                <path stroke-linecap="round" stroke-linejoin="round" d="M6 18L18 6M6 6l12 12" />
+              </svg>
+            </div>
+          </td>
+          <!--STATUS TEXT-->
+          <td class="basis 1/4">
+            <div v-if="user.status == 'FRIENDS'">
+              friends
+            </div>
+            <div v-else-if="user.status == 'NONE'">
+              click to add a friend
+            </div>
+            <div v-else-if="user.status == 'INVITATIONRECEIVED'">
+              friend request received
+            </div>
+            <div v-else-if="user.status == 'INVITATIONSENT'">
+              request status pending...
+            </div>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import {computed, inject, onMounted, ref} from "vue";
+import SortMenu from '@/components/SortMenu.vue'
+import * as _ from 'underscore';
+import type {GetUsersDto} from "@/dtos/GetUsersDto";
+import {UserService} from "@/services/UserService";
+
+const userService = inject<UserService>("userService");
+
+const searchInput = ref("");
+const sortType = ref("none"); //none ; friendStatus ; alphabetically
+const users = ref([]);
+
+userService?.getAllUsers()
+    .then((response) => {
+      users.value = response.data as GetUsersDto[];
+    })
+    .catch((error) => {
+      console.log("error: " + error);
+    });
+
+const filteredUsers = computed(()=> {
```

Comment:
I think it's a great use of computed method, we cache data. filtering is is also solved in a great way. nice and simple :)